### PR TITLE
Workbench: optional generator startup

### PR DESCRIPTION
**Pull request by deepfire**
Workbench changes:

1. add `--no-generator` to `start-cluster`, to skip the automatic `tx-generator` startup
2. add `--help` to `start-cluster`

### CODE REVIEW

File: nix/supervisord-cluster/default.nix [hunk 1]
```diff
@@ -134,20 +134,41 @@ let
...
+  startClusterUsage = ''
+Usage:
+   start-cluster [FLAGS..]
+
+   Flags:
+
+      --batch-name NAME               Override the batch name (default: ${batchName})
+      --no-generator | --no-gen       Don't auto-start the tx-generator
+
+      --trace | --debug               Trace the start-cluster script
```
Comment (MarcFontaine) [inline nix/supervisord-cluster/default.nix:146]: is there is difference between `--trace` and `--debug` ?

File: nix/workbench/run.sh [hunk 2]
```diff
@@ -289,7 +298,12 @@ EOF
         )
         genesis prepare "${genesis_args[@]}"
 
-        backend start-run "$dir" "$@"
+        ## Record genesis.
+        cp "$dir"/genesis/genesis.json "$dir"/genesis.json
```
Comment (MarcFontaine) [inline nix/workbench/run.sh:302]: Why do you need to do that ?
Why is this `cp` at this point in the script ?
IMHO having the same config twice in two files is confusing.

File: nix/supervisord-cluster/default.nix (hunk 1 above)
Comment (deepfire) [reply nix/supervisord-cluster/default.nix:146]: No, none.

File: nix/workbench/run.sh (hunk 2 above)
Comment (deepfire) [reply nix/workbench/run.sh:302]: It's necessary due to how genesis caching works.

File: nix/supervisord-cluster/default.nix (hunk 1 above)
Comment (MarcFontaine) [reply nix/supervisord-cluster/default.nix:146]: This confusing. There should not be two different flags that actually do the same thing.
Please choose one of the two.

**Review by MarcFontaine — APPROVED** [review]
Just merge it as it is

**Comment by deepfire** [comment]
We should work in increasing familiarity with the workbench inside the team -- including its concepts and principles.

It's easy to misinterpret things otherwise.

**Pull request merged**